### PR TITLE
Migrate connectors to use our python base image (Round 1)

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Migrate the following connectors to use our base image:
- source-sendgrid
- source-azure-blob-storage
- source-salesforce
- source-google-analytics-data-api

(Ignoring `source-s3` because we have to implement a `build_customization.py` following @flash1293 recent `nltk` changes to the `Dockerfile`.)

## How
Run 
```bash
airbyte-ci connectors --name=source-sendgrid --name=source-azure-blob-storage --name=source-salesforce --name=source-google-analytics-data-api migrate-to-base-image 31543
```

## 🚨 User Impact 🚨
None if the tests pass.

### CODE REVIEW

File: docs/integrations/sources/azure-blob-storage.md
Comment (bazarnov) [inline docs/integrations/sources/azure-blob-storage.md:181]: nit: "Replace Dockerfile with the python-connector-base image"?

**Comment by airbyte-oss-build-runner** [comment]
##  source-sendgrid test report (commit [`ae3c6214bc`](https://github.com/airbytehq/airbyte/pull/31543/commits/ae3c6214bcfe3810fd2bc5827a49be3ac306e1b5)) - ✅

⏲️  Total pipeline duration: 04mn42s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build source-sendgrid docker image for platform(s) linux/x86_64 | ✅       |
| Unit tests                                                      | ✅       |
| Acceptance tests                                                | ✅       |
| Code format checks                                              | ✅       |
| Validate metadata for source-sendgrid                           | ✅       |
| Connector version semver check                                  | ✅       |
| Connector version increment check                               | ✅       |
| QA checks                                                       | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_10-18-migrate_connectors_to_use_our_python_base_image_round_1/1697624165/ae3c6214bcfe3810fd2bc5827a49be3ac306e1b5/source-sendgrid/0.4.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:ae3c6214bcfe3810fd2bc5827a49be3ac306e1b5)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-sendgrid test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-azure-blob-storage test report (commit [`ae3c6214bc`](https://github.com/airbytehq/airbyte/pull/31543/commits/ae3c6214bcfe3810fd2bc5827a49be3ac306e1b5)) - ✅

⏲️  Total pipeline duration: 05mn35s 

| Step                                                                      | Result   |
|:--------------------------------------------------------------------------|:---------|
| Build source-azure-blob-storage docker image for platform(s) linux/x86_64 | ✅       |
| Acceptance tests                                                          | ✅       |
| Code format checks                                                        | ✅       |
| Validate metadata for source-azure-blob-storage                           | ✅       |
| Connector version semver check                                            | ✅       |
| Connector version increment check                                         | ✅       |
| QA checks                                                                 | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_10-18-migrate_connectors_to_use_our_python_base_image_round_1/1697624165/ae3c6214bcfe3810fd2bc5827a49be3ac306e1b5/source-azure-blob-storage/0.2.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:ae3c6214bcfe3810fd2bc5827a49be3ac306e1b5)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-azure-blob-storage test
```

**Comment by bazarnov** [comment]
@alafanechere You'll probably face the rate-limits for the `Google Analytics Data API` very shortly, it's recommended to have a separate PR for it, in order not to be blocked by the single connector and be able to merge timely.

**Comment by alafanechere** [comment]
> @alafanechere You'll probably face the rate-limits for the `Google Analytics Data API` very shortly, it's recommended to have a separate PR for it, in order not to be blocked by the single connector and be able to merge timely.

Thanks for the tips, will do if it gets annoying.

**Review by bazarnov — APPROVED** [review]
LGTM, with the small change to the changelog.

**Comment by airbyte-oss-build-runner** [comment]
##  source-salesforce test report (commit [`ae3c6214bc`](https://github.com/airbytehq/airbyte/pull/31543/commits/ae3c6214bcfe3810fd2bc5827a49be3ac306e1b5)) - ✅

⏲️  Total pipeline duration: 81mn55s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build source-salesforce docker image for platform(s) linux/x86_64 | ✅       |
| Unit tests                                                        | ✅       |
| Integration tests                                                 | ✅       |
| Acceptance tests                                                  | ✅       |
| Code format checks                                                | ✅       |
| Validate metadata for source-salesforce                           | ✅       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ✅       |
| QA checks                                                         | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_10-18-migrate_connectors_to_use_our_python_base_image_round_1/1697624165/ae3c6214bcfe3810fd2bc5827a49be3ac306e1b5/source-salesforce/2.1.5/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:ae3c6214bcfe3810fd2bc5827a49be3ac306e1b5)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-salesforce test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-google-analytics-data-api test report (commit [`ae3c6214bc`](https://github.com/airbytehq/airbyte/pull/31543/commits/ae3c6214bcfe3810fd2bc5827a49be3ac306e1b5)) - ✅

⏲️  Total pipeline duration: 85mn00s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Build source-google-analytics-data-api docker image for platform(s) linux/x86_64 | ✅       |
| Unit tests                                                                       | ✅       |
| Acceptance tests                                                                 | ✅       |
| Code format checks                                                               | ✅       |
| Validate metadata for source-google-analytics-data-api                           | ✅       |
| Connector version semver check                                                   | ✅       |
| Connector version increment check                                                | ✅       |
| QA checks                                                                        | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_10-18-migrate_connectors_to_use_our_python_base_image_round_1/1697624165/ae3c6214bcfe3810fd2bc5827a49be3ac306e1b5/source-google-analytics-data-api/2.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:ae3c6214bcfe3810fd2bc5827a49be3ac306e1b5)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-analytics-data-api test
```

**Comment by alafanechere** [comment]
@artem1205 the CI time for these connector did not get degraded. The image build duration remains around 30s.

**Comment by alafanechere** [comment]
I will merge this PR once https://github.com/airbytehq/airbyte/pull/31541 is approved and on master.

**Review by artem1205 — APPROVED** [review]

**Comment by airbyte-oss-build-runner** [comment]
##  source-sendgrid test report (commit [`328b0d1067`](https://github.com/airbytehq/airbyte/pull/31543/commits/328b0d106724dc7c6536c68d97f17340007ed46b)) - ❌

⏲️  Total pipeline duration: 04mn25s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build source-sendgrid docker image for platform(s) linux/x86_64 | ✅       |
| Unit tests                                                      | ✅       |
| Acceptance tests                                                | ✅       |
| Code format checks                                              | ✅       |
| Validate metadata for source-sendgrid                           | ❌       |
| Connector version semver check                                  | ✅       |
| Connector version increment check                               | ✅       |
| QA checks                                                       | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_10-18-migrate_connectors_to_use_our_python_base_image_round_1/1697636870/328b0d106724dc7c6536c68d97f17340007ed46b/source-sendgrid/0.4.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:328b0d106724dc7c6536c68d97f17340007ed46b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-sendgrid test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-azure-blob-storage test report (commit [`328b0d1067`](https://github.com/airbytehq/airbyte/pull/31543/commits/328b0d106724dc7c6536c68d97f17340007ed46b)) - ✅

⏲️  Total pipeline duration: 05mn15s 

| Step                                                                      | Result   |
|:--------------------------------------------------------------------------|:---------|
| Build source-azure-blob-storage docker image for platform(s) linux/x86_64 | ✅       |
| Acceptance tests                                                          | ✅       |
| Code format checks                                                        | ✅       |
| Validate metadata for source-azure-blob-storage                           | ✅       |
| Connector version semver check                                            | ✅       |
| Connector version increment check                                         | ✅       |
| QA checks                                                                 | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_10-18-migrate_connectors_to_use_our_python_base_image_round_1/1697636870/328b0d106724dc7c6536c68d97f17340007ed46b/source-azure-blob-storage/0.2.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:328b0d106724dc7c6536c68d97f17340007ed46b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-azure-blob-storage test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-google-analytics-data-api test report (commit [`328b0d1067`](https://github.com/airbytehq/airbyte/pull/31543/commits/328b0d106724dc7c6536c68d97f17340007ed46b)) - ❌

⏲️  Total pipeline duration: 87mn04s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Build source-google-analytics-data-api docker image for platform(s) linux/x86_64 | ✅       |
| Unit tests                                                                       | ✅       |
| Acceptance tests                                                                 | ✅       |
| Code format checks                                                               | ✅       |
| Validate metadata for source-google-analytics-data-api                           | ❌       |
| Connector version semver check                                                   | ✅       |
| Connector version increment check                                                | ✅       |
| QA checks                                                                        | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_10-18-migrate_connectors_to_use_our_python_base_image_round_1/1697636870/328b0d106724dc7c6536c68d97f17340007ed46b/source-google-analytics-data-api/2.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:328b0d106724dc7c6536c68d97f17340007ed46b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-analytics-data-api test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-google-analytics-data-api test report (commit [`d46cd29166`](https://github.com/airbytehq/airbyte/pull/31543/commits/d46cd291669c084266b60c860bf66988aa33e21c)) - ❌

⏲️  Total pipeline duration: 21mn43s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Build source-google-analytics-data-api docker image for platform(s) linux/x86_64 | ✅       |
| Unit tests                                                                       | ✅       |
| Acceptance tests                                                                 | ❌       |
| Code format checks                                                               | ✅       |
| Validate metadata for source-google-analytics-data-api                           | ✅       |
| Connector version semver check                                                   | ✅       |
| Connector version increment check                                                | ✅       |
| QA checks                                                                        | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_10-18-migrate_connectors_to_use_our_python_base_image_round_1/1697643708/d46cd291669c084266b60c860bf66988aa33e21c/source-google-analytics-data-api/2.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:d46cd291669c084266b60c860bf66988aa33e21c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-analytics-data-api test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-salesforce test report (commit [`d46cd29166`](https://github.com/airbytehq/airbyte/pull/31543/commits/d46cd291669c084266b60c860bf66988aa33e21c)) - ✅

⏲️  Total pipeline duration: 26mn42s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build source-salesforce docker image for platform(s) linux/x86_64 | ✅       |
| Unit tests                                                        | ✅       |
| Integration tests                                                 | ✅       |
| Acceptance tests                                                  | ✅       |
| Code format checks                                                | ✅       |
| Validate metadata for source-salesforce                           | ✅       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ✅       |
| QA checks                                                         | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_10-18-migrate_connectors_to_use_our_python_base_image_round_1/1697643708/d46cd291669c084266b60c860bf66988aa33e21c/source-salesforce/2.1.5/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:d46cd291669c084266b60c860bf66988aa33e21c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-salesforce test
```

**Comment by alafanechere** [comment]
Merging as `source-google-analytics-data-api` CAT failure is transient (probably due to a rate limiting issue): it passed two times in a row and failed on the third run (without any code change).

**Pull request merged**
